### ISSUE TITLE
context.fail! is confusing, behaviour has changed, docs haven't

### ISSUE THREAD

**Issue by matthucke** [collectiveidea/interactor#90]
In Version 3, context.fail! always throws an exception:

``` ruby
    # Raises Interactor::Failure initialized with the Interactor::Context.
    def fail!(context = {})
      modifiable.update(context)
      @failure = true
      raise Failure, self
    end
```

In version 2, it did not.  The README is still showing the version 2 behaviour, where the caller queries success? after.

What is the current best practice to indicate and respond to failure?  I liked the old idiom of checking context.success? from the controller, after the interactor has returned.  Yet now it seems we must rescue instead - a change that disrupts the structure of the controller methods.

Could we provide a non-throwing 'fail' method in addition to the throwing version?  (I'm currently using a Rails initializer to monkey-patch one in).

**Comment by laserlemon** [comment]
Are you using the class method `call` or the instance method?

**Comment by matthucke** [comment]
Neither.  I was testing internal business-logic methods of my interactor directly - before writing a `call` that would combine them in sequence.

Reading the source, I discover that `call` swallows exceptions from `fail!`, while `call!` does not.  

I'll submit a modification to the README that clarifies this.

**Comment by laserlemon** [comment]
Merged #91.
